File: frontend/src/api/tracking.ts
import apiClient from './client';

export interface TrackingStatus {
    tracking_id: string;
    status: string;
    origin: string;
    destination: string;
    vehicle_plate: string;
    driver_name: string;
    cargo_weight: number;
    distance_km: number | null;
    start_time: string | null;
    end_time: string | null;
}

export interface ShipmentSummary {
    tracking_id: string;
    origin: string;
    destination: string;
    status: string;
    start_time: string | null;
}

export const getTrackingStatus = async (trackingId: string): Promise<TrackingStatus> => {
    const response = await apiClient.get<TrackingStatus>(`/tracking/${trackingId}`);
    return response.data;
};

export const getMyShipments = async (): Promise<ShipmentSummary[]> => {
    const response = await apiClient.get<ShipmentSummary[]>('/tracking/my-shipments');
    return response.data;
};

File: backend/app/schemas/tracking.py
"""Pydantic schemas for Tracking endpoints."""

from pydantic import BaseModel


class TrackingResponse(BaseModel):
    tracking_id: str
    status: str
    origin: str
    destination: str
    vehicle_plate: str
    driver_name: str
    cargo_weight: int
    distance_km: float | None
    start_time: str | None
    end_time: str | None

    model_config = {"from_attributes": True}
